tests/functional/objc/primitives.d

```diff
@@ -3,6 +3,6 @@ extern (Objective-C):
 extern __gshared ObjcObject x9;
 extern __gshared Class x10;
 extern __gshared SEL x11;
-extern __gshared bool x12;
+extern __gshared BOOL x12;
```

Comment:
`BOOL` is `signed char`.

Reply:
Hmm, is `bool` guaranteed to be binary compatible with `signed char` in D?

Reply:
I have some problems with typedefs to long and unsigned long as they can be 32/64 bit... Windows tests are failing : |

Reply:
> Hmm, is bool guaranteed to be binary compatible with signed char in D

They're both the same size. But it's not guaranteed since the size of `char` (in C) is not guaranteed.

> I have some problems with typedefs to long and unsigned long as they can be 32/64 bit... Windows tests are failing : |

Of course :(.

Reply:
> But it's not guaranteed since the size of char (in C) is not guaranteed.

But it's de facto 1 byte, so I guess we can add:

tuple("BOOL", CXTypeKind.uChar): "bool",
tuple("BOOL", CXTypeKind.sChar): "bool",

to the table? Or not?